lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -21,27 +54,75 @@ class PokemonOverviewPage extends StatelessWidget {
         centerTitle: true,
         backgroundColor: primaryColor,
       ),
-      body: pokemons.when(
-        loading: () => const Center(child: CircularProgressIndicator()),
-        error: (errorMessage) {
-          WidgetsBinding.instance.addPostFrameCallback((_) => _showErrorMessageSnackbar(context, errorMessage));
-          return const Center(child: Text(noPokemonsAvailableLabel));
-        },
-        (data) => GridView.builder(
-          padding: const EdgeInsets.all(10.0),
-          itemCount: data.length,
-          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
-          itemBuilder: (_, index) {
-            final pokemon = data[index];
-            return PokemonCard(pokemon: pokemon);
-          },
-        ),
+      body: Column(
+        children: [
+          Padding(
+            padding: const EdgeInsets.all(10.0),
+            child: TextField(
+              controller: _searchController,
+              decoration: InputDecoration(
+                iconColor: primaryColor,
+                prefixIcon: const Icon(Icons.search),
+                suffixIcon: _searchController.text.isNotEmpty
+                    ? IconButton(
+                        onPressed: () {
+                          _onClearTextField();
+                          if (_searchController.text.isNotEmpty) return;
+                          _onDisposeSearchedPokemons();
+                        },
+                        icon: const Icon(Icons.close_rounded),
+                      )
+                    : null,
+              ),
+            ),
+          ),
+          Expanded(
+            child: _searchController.text.isNotEmpty
+                ? widget.searchedPokemons.isEmpty
+                    ? const Center(child: Text(noPokemonSearchResult))
+                    : GridView.builder(
+                        padding: const EdgeInsets.all(10.0),
+                        itemCount: widget.searchedPokemons.length,
+                        gridDelegate:
+                            const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: pokemonColumnCount),
+                        itemBuilder: (_, index) {
+                          final pokemon = widget.searchedPokemons[index];
+                          return PokemonCard(pokemon: pokemon);
+                        },
+                      )
+                : widget.pokemons.when(
+                    loading: () => const LoadingIndicator(),
+                    error: (errorMessage) {
+                      WidgetsBinding.instance.addPostFrameCallback((_) => _showErrorMessageSnackbar());
```

Comment:
why aren't you using the `errorMessage` lambda property above to pass and use for this snackbar callout? 🤔